File: src/main/java/de/steklopod/sec04/Lec04LimitRate.java
package de.steklopod.sec04;

import de.steklopod.utils.Util;
import reactor.core.publisher.Flux;

public class Lec04LimitRate {

    public static void main(String[] args) {

        Flux.range(1, 1000) // 175
                .log()
                .limitRate(100, 0) // 75%
                .subscribe(Util.subscriber());



    }


}

File: src/main/kotlin/de/steklopod/sec04/Lec04LimitRate.kt
package de.steklopod.sec04

import de.steklopod.utils.Util.subscriber
import reactor.core.publisher.Flux

object Lec04LimitRate {
    @JvmStatic
    fun main(args: Array<String>) {
        Flux.range(1, 1000) // 175
            .log()
            .limitRate(100, 0) // 75%
            .subscribe(subscriber())
    }
}
